cds_migrator_kit/circulation/items/api.py

```diff
@@ -0,0 +1,210 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015-2018 CERN.
+#
+# cds-migrator-kit is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""CDS Migrator Circulation API."""
+
+from datetime import datetime
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class LibrariesMigrator():
+    """Migrate legacy libraries to Invenio ILS records.
+
+    Expected input format for legacy libraries:
+        [
+          ...,
+          {
+            'id': ,
+            'name': ,
+            'address': ,
+            'email': ,
+            'phone': ,
+            'type': ,
+            'notes': ,
+          },
+          ...,
+        ]
+
+    """
+
+    IGNORE_LIBRARY_IDS = [
+        43,  # name Other
+    ]
+    VALID_LIBRARY_TYPES = ['main', 'internal']
+
+    def __init__(self, libraries):
+        """Constructor."""
+        self.libraries = [l for l in libraries
+                          if l['type'] in self.VALID_LIBRARY_TYPES and
+                          l['id'] not in self.IGNORE_LIBRARY_IDS]
+
+    def _migrate_internal_locations(self, location_pid):
+        """Return new internal location records."""
+        internal_locations = []
+        for i, library in enumerate(self.libraries):
+            internal_library = {
+                'ilocid': i,
+                'locid': location_pid,
+                'legacy_id': library['id'],
+                'name': library['name'],
+                'address': library.get('address', ''),
+                'email': library.get('email', ''),
+                'phone': library.get('phone', ''),
+                'notes': library.get('notes', ''),
+            }
+            internal_locations.append(internal_library)
+        return internal_locations
+
+    def migrate(self):
+        """Return location and internal location records."""
+        location_pid = '1'
+        internal_locations = self._migrate_internal_locations(location_pid)
+
+        location = {
+            'locid': '1',
+            'name': 'CERN Central Library',
+        }
+
+        return location, internal_locations
+
+
+class ItemsMigrator():
+    """Migrate legacy items to Invenio ILS records.
+
+    Expected input format for legacy items:
```

Comment:
the Item structure is a bit different from the one that we have received form the Library: [Item schema](https://github.com/CERNDocumentServer/cds-dojson/blob/books/cds_dojson/schemas/records/books/ymls/item.yml), I think we should discuss about it's final form

Reply:
Well, that one is not really complete and corresponding to the current data, for example:
* DOI does not make sense in items
* persistent_identifiers is not going to be like this
* barcode is 2 times (probably the second is location, but we decided to change the name)
* library does not correspond to ILS/Circulation

In any case, this is a first pass to have something done. There will be for sure changes in the future.
I am also updating the corresponding schemas in `invenio-app-ils`, I will also in `cds-dojson` to keep everything aligned, even if there is too much confusion with schemas scattered around.